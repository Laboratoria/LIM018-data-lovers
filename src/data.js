export function showFilms(dataFilms) {
  const filmCards = document.getElementById("filmCards") //<-- Llamando al contenedor de html
  filmCards.innerHTML='';
  dataFilms.forEach(film => {
    filmCards.innerHTML+= getFilmDiv(film)
  })
}

export function getFilmDiv(film) { //<--- Creando divs para las peliculas
  return `<div class="card">
      <img src="${film.poster}" alt="imagen">
      <img id="star" src="images/estrellita.png">
      <p id="score">${film.rt_score}</p>
      <h2>${film.title}</h2>
      <p>${film.release_date}</p>
      </div>`
 }

 export function sortBy (films) {
  return films.sort(function(a,b){
    return b.rt_score - a.rt_score
  })
 }

 export function sortAz (films) {
  return films.sort((a,b) => {
    if (a.title > b.title) {return 1}
    if (a.title < b.title) {return -1}
    return 0;
  });
 }

 export function sortZa (films) {
  return films.sort((a,b) => {
    if (a.title > b.title) {return -1}
    if (a.title < b.title) {return 1}
    return 0;
  });
 }

 export function sortLessOld (films) {
  return films.sort(function(a,b){
    return b.release_date - a.release_date
  })
 }
 
 export function sortOldest (films) {
  return films.sort(function(a,b){
    return a.release_date - b.release_date
  })
 }

 
 
