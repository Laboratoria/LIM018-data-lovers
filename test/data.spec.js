import { createCard, anotherExample } from '../src/data.js';


describe('createCard', () => {
  it('is a function', () => {
    expect(typeof createCard).toBe('function');
  });

  it('returns string', () => {
    const pokemon = {
      "num": "001",
      "name": "bulbasaur",
      "generation": {
        "num": "generation i",
        "name": "kanto"
      },
      "about": "Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sun's rays, the seed grows progressively larger.",
      "img": "https://www.serebii.net/pokemongo/pokemon/001.png",
      "size": {
        "height": "0.71 m",
        "weight": "6.9 kg"},
      "type":[
          "grass",
          "poison"
      ]}
    expect(createCard(pokemon)).toMatchSnapshot()
  
  });
});


describe('anotherExample', () => {
  it('is a function', () => {
    expect(typeof anotherExample).toBe('function');
  });

  it('returns `anotherExample`', () => {
    expect(anotherExample()).toBe('OMG');
  });
});
