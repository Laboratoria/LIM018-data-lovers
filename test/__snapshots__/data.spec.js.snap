// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCard returns string 1`] = `
"
     <section class=\\"info-card-pokemon\\"> 
        <div class=\\"img-pokemon\\">
         <img src=\\"https://www.serebii.net/pokemongo/pokemon/001.png\\">
       </div>
       <div class=\\"data-pokemon\\">   
         <span>bulbasaur</span>
         <span>001</span>
         <span>Tipo: grass, poison</span>
       </div>
     </section>
   "
`;
